### PR TITLE
[web-browser][ios] Add `modalPresentationStyle` option

### PR DESCRIPTION
**Pull request by barthap**
# Why

Feature requested by users.

Resolves ENG-4527

Adds a possibility for users to workaround issues like #13844

# How

Added the `presentationStyle` option ([iOS docs](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621355-modalpresentationstyle))

- Stole the enum from `expo-image-picker` - it is identical
- Fixed a bug where browser wasn't dismissed on swipe-down. I did it the same way as in #15511 (see that PR for a detailed description).

# Test Plan

NCL: Opened & dismissed browser with each enum value

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by tsapeta — APPROVED** [review]
Nice! Thanks for taking care of that 🙏

**Review by bbarthec — APPROVED** [review]
Looking neat 👍

File: packages/expo-web-browser/src/WebBrowser.types.ts [hunk 1]
```diff
@@ -124,6 +131,18 @@ export enum WebBrowserResultType {
   LOCKED = 'locked',
 }
 
+// @docsMissing
```
[the remark this answers is not in the document]
Comment (Simek) [reply packages/expo-web-browser/src/WebBrowser.types.ts:134]: I'm not sure if this applies here, but recently there was discussion about enums naming convention on TS side, and the conclusion was that we should stick to uppercase format:
* https://www.notion.so/expo/Expo-Modules-Styleguide-02e6caa7c76e45d2a9bf1e8113e4ce1a

[the remark this answers is not in the document]
Comment (barthap) [reply packages/expo-web-browser/src/WebBrowser.types.ts:134]: You're right 🙄 and the enum values should be strings. The `expo-image-picker` needs this fix too

File: packages/expo-web-browser/src/WebBrowser.types.ts [hunk 2]
```diff
@@ -124,6 +131,48 @@ export enum WebBrowserResultType {
...
+   * A presentation style where the browser is displayed over the app's content.
+   */
+  OVER_CURRENT_CONTEXT = 'overCurrentContext',
+  /**
+   * A presentation style where the browser is displayed in a popover view.
+   */
+  POPOVER = 'popover',
+  /**
+   * **iOS 13+:** The default presentation style chosen by the system.
+   * On older iOS versions, falls back to `WebBrowserPresentationStyle.FullScreen`.
```
Comment (Simek) [inline packages/expo-web-browser/src/WebBrowser.types.ts:171]:
```suggestion
   * @platform ios 13+ 
   * The default presentation style chosen by the system.
   * On older iOS versions, falls back to `WebBrowserPresentationStyle.FullScreen`.
```

Let's use annotation to mark required iOS version.

Comment (barthap) [reply packages/expo-web-browser/src/WebBrowser.types.ts:171]: Thanks! I missed that point in the `Expo modules style guide` 😅

**Pull request merged**